src/main/java/com/amazon/reviews/service/impl/CsvFileParsingServiceImpl.java

```diff
@@ -0,0 +1,76 @@
+package com.amazon.reviews.service.impl;
+
+import com.amazon.reviews.model.dto.ReviewDto;
+import com.amazon.reviews.service.FileParsingService;
+import com.amazon.reviews.service.FileReaderService;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+
+public class CsvFileParsingServiceImpl implements FileParsingService {
+    private static final int ID_INDEX = 0;
+    private static final int PRODUCT_ID_INDEX = 1;
+    private static final int USER_ID_INDEX = 2;
+    private static final int PROFILE_NAME_INDEX = 3;
+    private static final int HELPFULNESS_NUMERATOR_INDEX = 4;
+    private static final int HELPFULNESS_DENOMINATOR_INDEX = 5;
+    private static final int SCORE_INDEX = 6;
+    private static final int TIME_INDEX = 7;
+    private static final int SUMMARY_INDEX = 8;
+    private final static int TEXT_INDEX = 9;
+    private final static int LAST_CSV_COMMA = 9;
+    private final static int PARSED_LINE_SIZE = 10;
+    private final FileReaderService fileReaderService;
+
+    public CsvFileParsingServiceImpl(FileReaderService fileReaderService) {
+        this.fileReaderService = fileReaderService;
+    }
+
+    @Override
+    public List<ReviewDto> parse(String path) {
+        List<String> input = fileReaderService.read(path);
+        List<String[]> parsedInput = parseData(input);
+        List<ReviewDto> reviewDtos = formReviewDtos(parsedInput);
+        return reviewDtos;
+    }
+
+    private List<String[]> parseData(List<String> input) {
+        List<String[]> parsedLines = new ArrayList<>();
+        String reviewDetails;
+        String reviewText;
+        int commaIndex;
+        for (int i = 1; i < input.size(); i++) {
+            String[] parsedLine = new String[PARSED_LINE_SIZE];
+            String line = input.get(i);
+            commaIndex = StringUtils.ordinalIndexOf(line, ",", LAST_CSV_COMMA);
+            reviewDetails = line.substring(0, commaIndex);
+            reviewText = line.substring(commaIndex + 1);
+            String[] dataItems = reviewDetails.split(",");
+            for (int j = 0; j < PARSED_LINE_SIZE - 1; j++) {
+                parsedLine[j] = dataItems[j];
+            }
+            parsedLine[TEXT_INDEX] = reviewText;
+            parsedLines.add(parsedLine);
```

Comment:
can we use some external library to do this? our `text` field may (and will) contain commas

Reply:
I know we can use OpenCSV (or one of the other libraries), but every time I've used them, the `text` field is cut on the first comma, and I can't find a way to make OpenCsv ignore commas in the `text`. This is why I wrote this bulky code - it ignores commas in the `text` part and parses the CSV correctly. If you can help me set up OpenCSV or another library to work properly with the text field, I'd very grateful 🙏